Ayush/dev-744

hyusap opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced file upload support in chat with drag-and-drop, file selection, validation, and removal.
  - Added animated, collapsible "thinking" UI displaying AI thought streams and associated query/response data.
  - Redesigned chat layout with resizable panels and a mobile sidebar overlay.
  - Added new UI components: context menus, dropdown menus, alert dialogs, tooltips, scroll areas, and resizable panels.
  - Enhanced chat input with a modular prompt input component supporting rich interactions.
  - Added reaction management hook and improved AI message display with richer metadata and new streaming chunk types.
  - Added detailed hooks for reaction and thought management to improve message interactions.
  - Introduced timestamp formatting and right-click context menu in conversation tabs.
  - Added file icon display for uploaded files in messages.
  - Added documentation and rules instructing use of pnpm for package management.
  - Added a `Header` component to the settings layout for consistent UI.

- **Improvements**
  - Updated UI styling including buttons, inputs, fonts, and theme colors for better visual consistency.
  - Sidebar and user info sections restyled with improved dialogs and dropdown menus replacing modal popups.
  - Optimistic UI updates and error handling improved with toast notifications replacing modal alerts.
  - Streamlined message list by splitting user and AI messages into separate components and removing thought logic.
  - Enhanced streaming thought processing with incremental parsing and structured data updates.
  - Improved streaming thought parsing to separate thought, Honcho, and PDF queries in real-time.
  - Added runtime guard to skip Stripe webhook processing if Stripe secret key is missing.
  - Removed sticky positioning from header for layout consistency.
  - Updated prompt instructions and AI streaming logic for improved query formatting.
  - Refined Next.js configuration with typed settings and async headers.

- **Bug Fixes**
  - Added runtime guard to skip Stripe webhook processing if Stripe secret key is missing.
  - Removed console logging in cache miss scenarios for cleaner output.

- **Chores**
  - Added multiple Radix UI, animation, icon, and utility dependencies to support new UI components and features.
  - Updated package scripts to suppress Turbopack warnings during development.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

hyusap committed:
test the streaming

hyusap committed:
stuff i guess

hyusap committed:
Refactor Chat Component and Enhance Message Handling

- Updated the Chat component to utilize a new ThinkingData structure for managing thought content and responses.
- Removed legacy thought data handling and integrated new state management for streaming messages.
- Enhanced message processing in the getMessages function to include thinking data for AI messages.
- Updated the AIMessage component to conditionally render a ThinkBox based on the presence of thinking data.
- Refactored the ThinkBox component to support both streaming and historical message content.
- Removed unused Typing component and cleaned up related code.

This refactor improves the clarity and efficiency of message handling in the chat interface.

hyusap committed:
Enhance Chat Component with New UI Elements and Improved Functionality

- Added Radix UI components for alert dialogs, context menus, and dropdowns to enhance user interactions.
- Integrated the Sonner library for toast notifications, replacing SweetAlert2 for better user feedback.
- Refactored the Chat component to include responsive design features and improved state management for mobile views.
- Updated the Sidebar component to support conversation editing and deletion with confirmation dialogs.
- Enhanced file upload handling and added new UI elements for better user experience in message composition.
- Improved overall code structure and readability by removing unused imports and optimizing component logic.

These changes significantly improve the usability and responsiveness of the chat interface.

hyusap committed:
Refactor Sidebar Component to Use Dropdown Menu for Settings

- Replaced the previous settings button implementation with a DropdownMenu for improved user experience.
- Integrated new dropdown components from the UI library to streamline the settings and sign-out options.
- Removed unused state management related to the settings menu, enhancing code clarity and performance.

These changes enhance the Sidebar's functionality and maintainability.

hyusap committed:
Refactor Chat and Message Handling for Improved File Uploads and Parsing

- Updated the Chat component to enhance mobile responsiveness and improve state management for sidebar visibility.
- Modified message processing to include file uploads, appending file names to messages when applicable.
- Refactored UserMessage component to display uploaded files using a new FileUpload component.
- Replaced legacy XML parsing with a new delimiter-based approach for processing thought data, improving clarity and efficiency.
- Removed the StreamingXmlParser as part of the transition to the new parsing method.

These changes enhance the user experience by streamlining message handling and improving file upload functionality.

hyusap committed:
fix build errors

hyusap committed:
clear out debugging

hyusap commented:
@coderabbitai review

hyusap committed:
Update CLAUDE.md

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

hyusap committed:
Refactor Layout Components to Manage Header Visibility

- Removed the Header component from the RootLayout to streamline the layout structure.
- Added the Header component to the SettingsLayout for consistent navigation across settings pages.

These changes improve the layout organization and enhance user navigation experience in the settings section.

hyusap committed:
Remove Testing Page Component

- Deleted the Testing page component along with its associated logic and styles. This cleanup helps streamline the codebase by removing unused components and improving maintainability.

hyusap committed:
Refactor File Selection Handling in Chat Component

- Updated the Chat component to clear selected files immediately after appending them to the message, preventing re-attachment.
- Ensured selected files are cleared in both successful and error cases to maintain consistent state management.

These changes enhance the user experience by improving file handling during message composition.

hyusap committed:
Refactor Supabase Client Initialization in Chat Component

- Moved the Supabase client and fetchUser function outside the Chat component to prevent recreation on every render, improving performance and efficiency.
- This change enhances the overall responsiveness of the Chat component by optimizing user data fetching.

hyusap committed:
Add Hydration State Management to Chat Component

- Introduced a new state variable, isHydrated, to track the hydration status of the Chat component.
- Updated the rendering logic to conditionally display the user's full name based on the hydration state, enhancing the user experience during initial loading.

These changes improve the responsiveness and clarity of user information in the chat interface.

hyusap committed:
Improve PDF Query Handling in AI Response Function

- Updated the condition for checking the PDF query to use `trim()` and `toLowerCase()` methods, ensuring more robust handling of user input.
- This change prevents potential issues with empty or improperly formatted PDF queries, enhancing the reliability of the response generation process.

hyusap committed:
Refactor Thought Processing in AI Response Function

- Enhanced the handling of thought segments by splitting the input on the delimiter and processing each segment individually.
- Improved the logic to update the current section after each delimiter, ensuring more accurate state management during response generation.
- This refactor increases the clarity and efficiency of the thought processing mechanism, leading to better response handling.

hyusap committed:
Enhance Tooltip Component to Support Children Prop

- Updated the Tooltip and TooltipTrigger components to accept and render children, allowing for more flexible content within tooltips.
- Adjusted the TooltipContent className to use a more appropriate syntax for the transform-origin property, improving styling consistency.

These changes enhance the usability and customization of the tooltip components.

hyusap commented:
@coderabbitai review

hyusap committed:
Merge branch 'main' into ayush/dev-744

hyusap committed:
Refactor Thinking Data Update Logic in Chat Component

- Introduced a new helper function, updateThinkingData, to streamline the process of updating thinking data based on user input.
- Replaced inline logic for updating thinking data in message handling with calls to the new helper function, improving code readability and maintainability.

These changes enhance the clarity of the Chat component's logic and promote code reuse.

hyusap committed:
Enhance AIMessage Component with Clipboard Functionality

- Added state management for clipboard copy status, allowing users to see a confirmation when content is copied.
- Updated the copy button's appearance and tooltip text based on the copy status, improving user feedback.
- Refactored the copy to clipboard function to be asynchronous, ensuring a smoother user experience.

These changes enhance the interactivity and usability of the AIMessage component.

hyusap committed:
Refactor Chat Component for Improved Readability and Functionality

- Reformatted the updateThinkingData function for better readability by aligning conditional logic.
- Added an onNewChat handler to the ResizablePanel component to manage mobile sidebar state more effectively.
- Cleaned up unnecessary whitespace in the main rendering logic, enhancing code clarity.

These changes improve the maintainability and user experience of the Chat component.

hyusap committed:
Enhance File Upload Functionality in Chat Component

- Integrated a new FileUploadComponent to streamline file display and management within the Chat component.
- Added a helper function to parse file details, improving the handling of uploaded files.
- Updated the rendering logic to utilize the new component, enhancing code clarity and user experience.

These changes improve the file handling capabilities and overall maintainability of the Chat component.

VVoruganti reviewed (commented):
<img width="646" alt="image" src="https://github.com/user-attachments/assets/4dfa36ff-b1ae-417f-8af3-402a49fb760a" />

<img width="353" alt="image" src="https://github.com/user-attachments/assets/a47dde90-2c79-4b5e-99e9-762506e3eba0" />


Dark mode seems to be broken for me. A lot of the text is still black and having to highlight to see it, also don't see a manual toggle

hyusap committed:
colors and stuff

hyusap committed:
fix weird colors

hyusap committed:
Update package.json and pnpm-lock.yaml to include import-in-the-middle version 1.14.0

- Added import-in-the-middle as a dependency in package.json.
- Updated pnpm-lock.yaml to reflect the new version and its integrity hash.

These changes ensure the project uses the latest version of the import-in-the-middle package, improving compatibility and functionality.

hyusap committed:
type the file

hyusap committed:
move to turbopack for localdev

hyusap commented:
@VVoruganti fixed

hyusap committed:
fix: remove turbopack

hyusap committed:
Update Sentry configuration to disable source maps

- Changed the Sentry configuration to use an object for sourcemaps, setting disable to true. This enhances security by preventing source maps from being included in client bundles.
